react/features/participants-pane/components/native/VisitorsItem.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+
+import Button from '../../../base/ui/components/native/Button';
+import { BUTTON_TYPES } from '../../../base/ui/constants.native';
+import { approveRequest, denyRequest } from '../../../visitors/actions';
+import { IPromotionRequest } from '../../../visitors/types';
+
+import ParticipantItem from './ParticipantItem';
+import styles from './styles';
+
+interface IProps {
+
+    /**
+     * Promotion request reference.
+     */
+    request: IPromotionRequest;
+}
+
+export const VisitorsItem = ({ request: r }: IProps) => {
+    const dispatch = useDispatch();
+    const admit = useCallback(() => dispatch(approveRequest(r)), [ dispatch ]);
+    const reject = useCallback(() => dispatch(denyRequest(r)), [ dispatch ]);
```

Comment:
Shouldn't we add the `r` in the list of dependencies next to `dispatch`? 

Since the request is an object we should be careful to not change the reference if there is no need for it.